### PR TITLE
feat/35: Add support for PACT_BROKER_WEBHOOK_RETRY_SCHEDULE environment variable

### PR DESCRIPTION
**Pull request by pitschr**
https://github.com/pact-foundation/pact-broker-docker/issues/35

### CODE REVIEW

File: pact_broker/docker_configuration.rb [hunk 1]
```diff
@@ -128,6 +132,35 @@ def to_s
...
+        }.collect { | word |
+          word.to_i
+        }
+        SpaceDelimitedIntegerList.new(array)
+      end
+
+      def to_s
+        collect do | integer |
+          integer.to_s
+        end.join(' ')
```
Comment (bethesque) [inline pact_broker/docker_configuration.rb:160]:
```suggestion
        collect(&:to_s).join(' ')
```

File: pact_broker/docker_configuration.rb [hunk 2]
```diff
@@ -128,6 +132,35 @@ def to_s
...
+      def initialize list
+        super(list)
+      end
+
+      def self.parse(string)
+        array = (string || '').split(' ').filter { |word|
+          (Integer(word) rescue nil) != nil
+        }.collect { | word |
+          word.to_i
+        }
```
Comment (bethesque) [inline pact_broker/docker_configuration.rb:153]:
```suggestion
        }.collect(&:to_i)
```

File: pact_broker/docker_configuration.rb [hunk 3]
```diff
@@ -128,6 +132,35 @@ def to_s
...
+    end
+
+    class SpaceDelimitedIntegerList < Array
+      def initialize list
+        super(list)
+      end
+
+      def self.parse(string)
+        array = (string || '').split(' ').filter { |word|
+          (Integer(word) rescue nil) != nil
```
Comment (bethesque) [inline pact_broker/docker_configuration.rb:150]: I'd personally pull this into an `integer?(string)` method.

**Review by bethesque — APPROVED** [review]
Looks great, thanks so much! I've made a few suggestions, but they're very nitpicky, and I'd just merge it if you didn't need to get the semantic commit test passing anyway.

**Comment by bethesque** [comment]
The semantic pull request plugin is complaining because the commit message doesn't start with "feat: ..." as per https://www.conventionalcommits.org/en/v1.0.0-beta.2/ If you amend your commit and change the message to "feat: add support for ..." then force push, it should be all 👍🏽

File: pact_broker/docker_configuration.rb (hunk 1 above)
Comment (pitschr) [reply pact_broker/docker_configuration.rb]: good advice. I was not aware of this method reference. Thanks.

File: pact_broker/docker_configuration.rb (hunk 3 above)
Comment (pitschr) [reply pact_broker/docker_configuration.rb]: I moved it to `self.integer?` method. However, for some reasons I could adjust the filter method to same style like `collect`, what I have tried are:
```
.filter(self::integer)
.filter(self:integer?)
.filter(integer?)
.filter(&:integer?)
```

So I accomplish it using as it was the only way I found that compiles the files & passes the tests:
```
.filter { |word| integer?(word) }
```

Let me know if you see another point for improvement

Comment (bethesque) [reply pact_broker/docker_configuration.rb]: Yeah, the `&:method` trick only works on the object being iterated over. What you've done is perfect.

**Comment by bethesque** [comment]
Thank you so much!

**Pull request merged**